src/marqo/tensor_search/tensor_search.py

```diff
@@ -417,12 +417,17 @@ def add_documents(config: Config, index_name: str, docs: List[dict], auto_refres
 
     existing_fields = set(index_info.properties.keys())
     new_fields = set()
+    # A dict to store the multimodal_fields and their sub_fields
+    # dict = {multi_modal_field_1 : set(),
+    #      =  multi_modal_field_2 : set()}
+    new_multimodal_combo_fields = dict()
```

Comment:
Question: How does the logic differ from `new_fields`?

Reply:
all the fields in the `new_fields` will be updated to `knn_fields_names` , add for `filtering`, and add to `text_properties`. however, for `multimodal_combo_fields`, `parent_field` will be added to `knn_fields_names`. but it  should not appear as a stand-alone field in `filtering` or `text_properties`.

Reply:
done